active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,67 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+Currently the ItemsSourceView class does not expose an IndexOf function. However during development, this method was used multiple times internally on derivatives of the class. To make the usage of this existing function easier on subclasses, we should expose this function in the ItemsSourceView.
+
+# Description
+
+The IndexOf function returns the index of the given element in the items source the ItemsSourceView is used for.
+Note that if an element is not contained in the items source, the function will return -1.
+
+# Examples
+C# example
+
+```c#
+/// ************** Behavior ************ ///
+
+var itemsSourceView = new ItemsSourceView()
+{
+  // Initialize
+};
+
+// Get an item
+var myItem = MyItemsSource.GetAt(1);
+
+Assert.AreEqual(1,itemsSourceView.IndexOf(myItem));
+
+/// ****** Use case in event handler ****** ///
+
+public void ItemInvoked(object sender, RoutedEventArgs e)
+{
+  int index = myItemsSourceView.IndexOf(sender as IInspectable); 
+
+  // Do something with index, here e.g. :
+  Console.log("Item " + index + " of items source was invoked");
+}
+
+```
+
+
+# Remarks
+
+The IndexOf function of ItemsSourceView base class throws a winrt hresult_not_implemented exceptions.
+If an element is not present in the collection, the method should return -1.
+
+# API Notes
+ItemsSourceView.IndexOf(IInspectable item): Returns index in the items source. If an item is not in the items source, it will return -1.
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```MIDL
+
+unsealed runtimeclass ItemsSourceView: Windows.UI.Xaml.Interop.INotifyCollectionChanged
+{
+
+    Int32 IndexOf(IInspectable item);
+
+}
+
+```
+
+# Appendix
+<!-- Anything else that you want to write down for posterity, but 
+that isn't necessary to understand the purpose and usage of the API.
+For example, implementation details. -->
+
+When IndexOf of the ItemsSourceView class get's invoked, should we instead of throwing `hresult_not_implemented` return -1 instead?
```

Comment:
Spec already says return -1, which I think is right.

Reply:
The scenario I want to cover is when one somehow get's a raw ItemsSourceView class (which I think is possible). And in that case, what should we do when someone invokes IndexOf on that raw class? Almost all functions just throw `hresult_not_implemented` . My question would be, whether we should do that or just return -1 in that case.

Reply:
I think [UIElementCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.UIElementCollection) and [List<T>](https://docs.microsoft.com/dotnet/api/System.Collections.Generic.List-1) provide great precedence for returning -1 rather than failing.

Reply:
Oh I see. In that case it may be better to align with those then. My reasoning is that virtually every other method regardless of return value just throws `not implemented`, so having different behaviors of "Hey I do not provide this feature" seems a bit confusing/incoherent for me. But there are valid reasons for both sides here, so I leave it up to you.